ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractKeysIT.scala

```diff
@@ -362,4 +362,31 @@ final class ContractKeysIT extends LedgerTestSuite {
         assertGrpcError(failedFetch, Status.Code.INVALID_ARGUMENT, "not visible")
       }
   })
+
+  test(
+    "CKDivulgedContractKeyReusability",
+    "Subsequent divulged contracts can use the same contract key",
+    allocate(SingleParty, SingleParty),
+  )(implicit ec => {
+    case Participants(Participant(ledger1, party1), Participant(ledger2, party2)) =>
+      for {
+        creator1 <- ledger1.create(party1, WithKeyCreator(party1, party2))
+        withKey1 <- ledger1.exerciseAndGetContract[WithKey](
+          party1,
+          creator1.exerciseWithKeyCreator_DiscloseCreate(_, party1),
+        )
+
```

Comment:
Maybe check that the contract is usable for interpretation here?

Reply:
Do you mean checking the structure of the Scala object we get? Or fetching the contract to be sure that it's ok?

Reply:
Submit a command on p2 that fetches the contract to check that it has been disclosed.

Reply:
Ok, thanks!

Reply:
@cocreature I've added additional verification that the disclosed contract can be fetched by the `party2`. I would appreciate your review :)

Reply:
Good stuff!

Reply:
Thanks!